pkg/segment/writer/segmetarw.go

```diff
@@ -93,31 +93,20 @@ func getSegFullMetaFnameFromSegkey(segkey string) string {
 
 func ReadSegmeta(smFilename string) []*structs.SegMeta {
 	smrLock.RLock()
-	retVal, err := getAllSegmetas(smFilename)
+	segMetas, err := getAllSegmetas(smFilename)
 	smrLock.RUnlock()
 	if err != nil {
 		log.Errorf("ReadSegmeta: getallsegmetas err=%v ", err)
 	}
-	return retVal
-}
-
-// read only the current node's segmeta
-func ReadLocalSegmeta(readFullMeta bool) []*structs.SegMeta {
 
-	smrLock.RLock()
-	retVal, err := getAllSegmetas(localSegmetaFname)
-	smrLock.RUnlock()
-	if err != nil {
-		log.Errorf("ReadLocalSegmeta: getallsegmetas err=%v ", err)
-		return retVal
-	}
+	readSfmForSegMetas(segMetas)
```

Comment:
Why do we need this inside `ReadSegmeta()`? (continuation of my previous comment)

Reply:
This is [present](https://github.com/siglens/siglens/pull/1744/files#diff-7f61f72d8e97fc004dc47f9ac98903e19cd28a4f4088c8d922cb86cfb76f9520) since the beginning. We need to set column names, PQIDs etc from sfm file into the current segMeta entries.